go/tasks/pluginmachinery/flytek8s/container_helper.go

```diff
@@ -22,21 +22,32 @@ const resourceGPU = "gpu"
 // Copied from: k8s.io/autoscaler/cluster-autoscaler/utils/gpu/gpu.go
 const ResourceNvidiaGPU = "nvidia.com/gpu"
 
-func MergeResources(in v1.ResourceRequirements, out *v1.ResourceRequirements) {
-	if out.Limits == nil {
-		out.Limits = in.Limits
-	} else if in.Limits != nil {
-		for key, val := range in.Limits {
-			out.Limits[key] = val
+func MergeResources(preferred v1.ResourceRequirements, resources v1.ResourceRequirements) v1.ResourceRequirements {
+	var result = v1.ResourceRequirements{
+		Limits:   make(v1.ResourceList),
+		Requests: make(v1.ResourceList),
+	}
+	for key, val := range resources.Limits {
+		result.Limits[key] = val
+	}
+	for key, val := range resources.Requests {
+		result.Requests[key] = val
+	}
+	if resources.Limits == nil {
+		result.Limits = preferred.Limits
```

Comment:
preferred here seems to be inverted?
it seems to get applied only when resources are nil. 
so should they be called `default` instead